src/components/app.jsx

```diff
@@ -60,6 +62,23 @@ class App extends React.Component{
       
     }
 
+    setActiveCard(index){
+       
+        if(typeof this.state.cards[0] !== 'undefined'){
```

Comment:
The `0` should be replaced with `index`. You are always checking the `0` index instead of checking if the index passed in is valid.

```JAVASCRIPT
if(typeof this.state.cards[index] !== 'undefined')
```